lib/test-collection.js

```diff
@@ -1,10 +1,42 @@
 'use strict';
 const EventEmitter = require('events');
+const cloneDeep = require('lodash.clonedeep');
 const fnName = require('fn-name');
 const Concurrent = require('./concurrent');
 const Sequence = require('./sequence');
 const Test = require('./test');
 
+class ContextBind {
+	constructor(ref) {
+		this.ref = ref;
+		this.bound = false;
+		this.boundContext = {};
+	}
+
+	get context() {
+		if (!this.bound) {
+			this.context = cloneDeep(this.ref.context);
```

Comment:
I find relying on the setter's behavior to be quite surprising. It's not immediately clear that `this.bound` and `this.boundContext` are actually updated.